### ISSUE TITLE
[BLES00153] The Orange Box (TF2)

### ISSUE THREAD

**Issue by Beyley** [RPCS3/rpcs3#13634]
### Quick summary

Orange Box TF2 freezes upon reaching the online connection screen

### Details

The Orange Box gets to the launcher
![image](https://user-images.githubusercontent.com/34501060/230789848-6e3cce8e-93f6-4fd0-98fd-ed9adbdf8836.png)
then selecting TF2 will cause the game to open, but shortly thereafter, it will freeze, trying to connect to the servers
![image](https://user-images.githubusercontent.com/34501060/230789881-ad3127d4-8972-4106-9d22-373ce26fcaa0.png)

What *should* happen, is the game fully fades in that dialog box, tries to make a DNS request, then attempts to connect to some server with SSL, fails, and shows a prompt to retry, but RPCS3 dies well before that, seemingly before even attempting to make the DNS request

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11186059/RPCS3.log.gz)

### System configuration

OS: Arch Linux
CPU: i9-12900
GPU: RX 6900XT (with RADV)
RAM: 64GB

RPCS3 version: 
![image](https://user-images.githubusercontent.com/34501060/230790219-631574e5-7962-4900-9f64-312d12f9a8ff.png)

**Comment by FlexBy420** [comment]
TF2 requires custom server to be made which is out of scope of RPCS3 project.

**Comment by Beyley** [comment]
> TF2 requires custom server to be made which is out of scope of RPCS3 project.

Yes, however it freezes before attempting to connect to said now-shut-down server, which *is* a bug, and it makes developing a reverse engineered server much harder, given you cant use RPCS3, and have to use a real console while developing

**Comment by jvyden** [comment]
> TF2 requires custom server to be made which is out of scope of RPCS3 project.

I'm currently working with @Beyley to develop these.

**Comment by Margen67** [comment]
Luctaris recently discovered you can get ingame (can still be unstable) by creating `<GAMEDIR>/USRDIR/GAME/TF/CFG/MODSETTINGS.CFG` with the following:
```
map_commentary ctf_2fort
exec CONFIG.PS3.CFG
```
![image](https://github.com/RPCS3/rpcs3/assets/3462541/cf026064-4a3e-4acd-936a-f4e2dc7f9262)
Maps without commentary will crash after a class is selected.
Pressing start makes the game try to reconnect to EA online again.

**Comment by Mrlinkwii** [comment]
how is this on latest release @Beyley  , since their has been many network updates since

**Comment by CoachSludge** [comment]
> how is this on latest release @Beyley , since their has been many network updates since

bumping this, RPCN has come along way, might try to figure it out myself.
my only other option is to add more maps via [srctools](https://github.com/ata4/bspsrc)

**Comment by frickinfire** [comment]
almost 2 years later this issue still persists, EA bypass @Margen67 provided _partially_ works, doesn't work with maps like 2fort, dustbowl, and granary

would love to see tf2 support improved, even with the bypass its pretty unstable and crashes often

**Comment by valters-tomsons** [comment]
RPCS3 does actually initiate connection to the backend, but gets stuck in a loop with the connection prompt flickering.  

On actual console, a loading screen appears and then transitions to main menu:

![Image](https://github.com/user-attachments/assets/4a2b7810-0bd1-4ef0-98bb-584f4836c269)

**Comment by valters-tomsons** [comment]
It no longer loops during connection, but gets stuck with a flickering menu and background. You can get it to function for a moment by pausing and resuming main thread a few times during connection.

<img width="1979" height="1219" alt="Image" src="https://github.com/user-attachments/assets/097c728f-aa10-478a-905d-a5807a546e40" />

https://github.com/user-attachments/assets/6249232d-71c2-4559-a636-0cb4975e7ace

**Comment by digant73** [comment]
please provide an update on this issue
